src/webapi/html_elements/select.rs

```diff
@@ -22,49 +39,45 @@ impl IElement for SelectElement {}
 impl IHtmlElement for SelectElement {}
 
 impl SelectElement {
-    /// Returns the `Some(index)` of the first selected item, if any, or `None` if there is no selected item.
-    // https://html.spec.whatwg.org/#the-select-element:dom-select-selectedindex
-    pub fn selected_index(&self) -> Option<i32> {
-        let si = js! (
-            return @{self}.selectedIndex;
-        ).try_into().unwrap();
-        if si < 0 {
-            None
-        } else {
-            Some(si)
-        }
+    /// Returns the value attribute of the first selected `<option>` element or
+    /// if it is missing, the text attribute. If there is no selection, return empty string.
+    /// This method is just a wrapper for getting `HTMLSelectElement.value` directly
+    // https://html.spec.whatwg.org/#the-select-element:dom-select-value
+    pub fn raw_value(&self) -> String {
+        js!(
+            return @{self}.value
+        ).try_into().unwrap()
     }
 
-    /// Change selected index to the given value.
-    // https://html.spec.whatwg.org/#the-select-element:dom-select-selectedindex
-    pub fn set_selected_index(&self, selected_index: Option<i32>) {
-        let selected_index = selected_index.unwrap_or(-1);
+    /// Set the given value to `HTMLSelectElement.value` directly.
+    // https://html.spec.whatwg.org/#the-select-element:dom-select-value
+    pub fn set_raw_value(&self, value: &str) {
         js!{
             @(no_return)
-            @{self}.selectedIndex = @{selected_index};
+            @{self}.value = @{value};
         }
     }
-
-    /// Returns the `Some(value)` of the first selected item, if any, or `None` if there is no selected item.
-    // https://html.spec.whatwg.org/#the-select-element:dom-select-value
-    pub fn value(&self) -> Option<String> {
-        js!(
+    
+    /// Returns the `Some(index)` of the first selected item, if any, or `None` if there is no selected item.
+    // https://html.spec.whatwg.org/#the-select-element:dom-select-selectedindex
+    pub fn selected_index(&self) -> Option<u32> {
+        js! (
             var self = @{self};
             if (self.selectedIndex < 0) {
                 return null;
-            } else {
-                return self.value;
+            }else{
+                return self.selectedIndex;
             }
         ).try_into().unwrap()
     }
 
-    /// Change the selected value to the given value.
-    // https://html.spec.whatwg.org/#the-select-element:dom-select-value
-    pub fn set_value(&self, value: Option<String>) {
-        match value{
-            Some(value) => js!{
+    /// Change selected index to the given value.
+    // https://html.spec.whatwg.org/#the-select-element:dom-select-selectedindex
+    pub fn set_selected_index(&self, selected_index: &Option<u32>) {
```

Comment:
Taking the `Option` by reference (`&`) is unnecessary.

Is there any particular reason why you made it so?